admin-dev/themes/new-theme/js/app/utils/datepicker.js

```diff
@@ -65,27 +67,28 @@ const replaceDatePicker = () => {
 const init = function initDatePickers() {
   const $datePickers = $('.datepicker input[type="text"]');
   $.each($datePickers, (i, picker) => {
-    $(picker).datetimepicker(
-      {
+    $(picker)
+      .datetimepicker({
         locale: window.full_language_code,
-        format: $(picker).data('format') ? $(picker).data('format') : 'YYYY-MM-DD',
+        format: $(picker).data('format')
+          ? $(picker).data('format')
+          : 'YYYY-MM-DD',
         sideBySide: true,
         icons: {
           time: 'time',
           date: 'date',
           up: 'up',
           down: 'down',
         },
-      },
-    ).on(
-      'dp.show',
-      replaceDatePicker,
-    ).on(
-      'dp.hide',
-      () => {
+      })
+      .on('dp.show', replaceDatePicker)
+      .on('dp.hide', () => {
         $(window).off('resize', replaceDatePicker);
-      },
-    );
+      })
+      .on('dp.change', (e) => {
+        const event = new CustomEvent('datepickerChange', e);
```

Comment:
A little comment to explain why we needed to add this might be helpful in the future